File: problems/helpers/linkedlist/ListNode.java
package problems.helpers.linkedlist;
import java.util.List;

public class ListNode {
    public int val;
    public ListNode next;
    public ListNode() {}
    public ListNode(int val) { this.val = val; }
    public ListNode(int val, ListNode next) { this.val = val; this.next = next; }
    
    static public ListNode createLinkedList(List<Integer> li) {
        ListNode root = new ListNode(li.get(0));
        ListNode next = root;
        for (int item : li.subList(1, li.size())) {
            next.next = new ListNode(item);
            next = next.next;
        }
        return root;
    }

    static public void printLinkedList(ListNode node) {
        while(node != null) {
            System.out.print(node.val + " ");
            node = node.next;
        }
        System.out.println("");
    }
}

File: problems/helpers/linkedlist/ListNode.swift
public class ListNode {
    public var val: Int
    public var next: ListNode?
    public init() { self.val = 0; self.next = nil; }
    public init(_ val: Int) { self.val = val; self.next = nil; }
    public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }

    static public func createListNode(_ li: [Int]) -> ListNode {
        let root = ListNode(li[0])
        var next = root
        for item in li[1...] {
            next.next = ListNode(item)
            next = next.next ?? ListNode()
        }
        return root
    }

    static public func printLinkedList(_ node: ListNode?) {
        if let listNode = node {
            print(listNode.val, terminator:" ")
            printLinkedList(listNode.next)
        } else {
            print("\n")
        }
    }
}